magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+from subprocess import check_call, check_output
+
+import ipcalc  # type: ignore[import]
+import yaml
+from systemd.journal import JournalHandler  # type: ignore[import]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(JournalHandler())
+logger.setLevel(logging.INFO)
+
+
+class AGWInstallerNetworkConfigurator:
+
+    DNS_ADDRESSES = "8.8.8.8 208.67.222.222"
+    INTERFACES_DIR = "/etc/network/interfaces.d"
+    REBOOT_NEEDED = False
```

Comment:
I think we should keep global variables for things that don't change. Reboot needed can be a class attribute (`self.reboot_needed = False`)

Reply:
Also, where is this `REBOOT_NEEDED` used anyway? I only see methods that change it to true but no logic around actually rebooting

Reply:
Changed.
Logic consuming this attribute is coming up in the next PR.